src/document/DocumentAnnotator.ts

```diff
@@ -60,14 +60,21 @@ export default class DocumentAnnotator extends BaseAnnotator {
         const pageReferenceEl = this.getPageReference(pageEl);
         const managers = this.managers.get(pageNumber) || new Set();
         const resinTags = { fileid: fileId, iscurrent: isCurrentFileVersion };
+        let destroyManagers = false;
 
-        // Destroy any managers that were attached to page elements that no longer exist
+        // If a referenced page element doesn't exist, destroy all managers to keep them in sync
```

Comment:
This might work a little better. We don't need to iterate through the whole list to determine if a certain element meets the criteria. What do you think about doing something like this  let destroyMangers = mangers.find(manager=>!manager.exists(pageEl))? destroyMangers will be undefined if the condition is not matched and you can use that in your conditional.

Reply:
Set() doesn't have find as far as I know. That's why we're limited to forEach. We would have to change it to an array, and that potentially affect performance. But then we could use `.some` which would look cleaner. But I don't think it's worth it.

Reply:
Ahh gotcha. That's a set not an array. Cool nm I didn't see that.